### PR TITLE
feature(Plugins/Premium Plugins/Spellchecker pro): Update docs for Spellchecker pro premium plugin, tinydrive and rtc

### PR DESCRIPTION
**Pull request by hamza0867**
…ellchecker pro premium plugin, tinydrive and rtc

Related Ticket: TINY-8595

Description of Changes:
This PR updates the docs for the Spellchecker pro premium plugin, the tiny drive premium plugin and etc premium plugin

Pre-checks:
- [x] Branch prefixed with `feature/` or `hotfix/`
- [x] `_data/nav.yml` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] Files removed have been deleted, not just excluded from the build (if applicable)
- [x] (New product features only) Release Note added

Review:
- [x] Documentation Team Lead has reviewed
- [x] Product Manager has reviewed

### CODE REVIEW

File: modules/ROOT/partials/configuration/spellchecker_dialog.adoc [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 This option specifies the primary spell checking mode.
 
-* When set to `true`, the <<toolbarbuttons,`spellcheck` toolbar button>> will open a dialog that will step the user through each potential spelling error in the document.
-* When set to `false`, the <<toolbarbuttons,`spellcheck` toolbar button>> will enable or disable as-you-type spell checking and the <<toolbarbuttons,`spellcheckdialog` toolbar button>> will open the spell checker dialog.
+* When set to `true`, the xref:toolbarbuttons[`spellcheck` toolbar button] will open a dialog that will step the user through each potential spelling error in the document.
+* When set to `false`, the xref:toolbarbuttons[`spellcheck` toolbar button] will enable or disable as-you-type spell checking and the xref:toolbarbuttons[`spellcheckdialog` toolbar button] will open the spell checker dialog.
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/spellchecker_dialog.adoc:6]:
```suggestion
* When set to `true`, the xref:plugins/premium/tinymcespellchecker.adoc#toolbarbuttons[`spellcheck` toolbar button] will open a dialog that will step the user through each potential spelling error in the document.
* When set to `false`, the xref:plugins/premium/tinymcespellchecker.adoc#toolbarbuttons[`spellcheck` toolbar button] will enable or disable as-you-type spell checking and the xref:plugins/premium/tinymcespellchecker.adoc#toolbarbuttons[`spellcheckdialog` toolbar button] will open the spell checker dialog.
```
`toolbarbuttons` doesn't exist in this file, so we need to make sure it links to where we expect.

File: modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc [hunk 2]
```diff
@@ -79,8 +79,6 @@ include::partial$configuration/spellchecker_select_languages.adoc[]
 
 // Bookmark for deprecated option name
 
-+++<a class="anchor" id="spellchecker_whitelist">++++++</a>+++
```
Comment (lnewson) [inline modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc:82]: This shouldn't have just been removed like this as we want this link to work as per the comment above. For example:
```
// Bookmark for deprecated spellchecker_whitelist option name
[[spellchecker_ignore_list]]
=== [[spellchecker_whitelist]] `spellchecker_ignore_list`
```
See: https://docs.asciidoctor.org/asciidoc/latest/sections/custom-ids/#assign-auxiliary-ids

- [ ] Add back this anchor in the `spellchecker_ignore_list.adoc` file and move the comment there as well

**Review by lnewson — COMMENTED** [review]
There's a broken `include` in the `content_langs` section on this page (see `modules/ROOT/partials/plugins/tinymcespellchecker-content-langs.adoc`):
image

- [x] Fix broken include

File: modules/ROOT/partials/configuration/spellchecker_ignore_list.adoc [hunk 3]
```diff
@@ -2,7 +2,7 @@
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/spellchecker_ignore_list.adoc:2]: This needs an id added to the option (all options should always have a fixed id).

File: modules/ROOT/partials/configuration/spellchecker_on_load.adoc [hunk 4]
```diff
@@ -1,6 +1,6 @@
 === `spellchecker_on_load`
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/spellchecker_on_load.adoc:1]:
```suggestion
[[spellchecker_on_load]]
=== `spellchecker_on_load`
```

File: modules/ROOT/partials/configuration/spellchecker_rpc_url.adoc [hunk 5]
```diff
@@ -1,8 +1,8 @@
 === `spellchecker_rpc_url`
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/spellchecker_rpc_url.adoc:1]:
```suggestion
[[spellchecker_rpc_url]]
=== `spellchecker_rpc_url`
```

File: modules/ROOT/partials/configuration/spellchecker_select_languages.adoc [hunk 6]
```diff
@@ -4,7 +4,7 @@ IMPORTANT: This option will be removed in {productname} 6.0.
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/spellchecker_select_languages.adoc:4]: This file needs an ID added to the option. e.g:
```
[[spellchecker_select_languages]]
=== `spellchecker_select_languages`
```

File: modules/ROOT/partials/configuration/spellchecker_select_languages.adoc [hunk 7]
```diff
@@ -4,7 +4,7 @@ IMPORTANT: This option will be removed in {productname} 6.0.
 
 NOTE: The `spellchecker_select_languages` option was deprecated with the release of {productname} 5.9. The `content_langs` option should be used instead. This change was made to allow more control over the way languages are specified, and to reflect that the option can be used outside of spell checking.
 
-This option specifies the languages that can be set for content when working with multi-language content. The specified languages will be available from the <<toolbarbuttons,`language` toolbar drop-down menu button>>.  For a list of available languages, see: <<supportedlanguages,Supported languages>>.
+This option specifies the languages that can be set for content when working with multi-language content. The specified languages will be available from the xref:toolbarbuttons[`language` toolbar drop-down menu button].  For a list of available languages, see: xref:supportedlanguages[Supported languages].
```
Comment (lnewson) [inline modules/ROOT/partials/configuration/spellchecker_select_languages.adoc:7]:
```suggestion
This option specifies the languages that can be set for content when working with multi-language content. The specified languages will be available from the xref:plugins/premium/tinymcespellchecker.adoc#toolbarbuttons[`language` toolbar drop-down menu button].  For a list of available languages, see: xref:plugins/premium/tinymcespellchecker.adoc#supportedlanguages[Supported languages].
```

File: modules/ROOT/partials/plugins/tinymcespellchecker-content-langs.adoc [hunk 8]
```diff
@@ -11,8 +11,8 @@ include::partial$configuration/content_langs_base.adoc[]
 The Spell Checker Pro plugin will use the `code` property to determine which dictionary to use on the content, unless the `customCode` property is available. The language tag specified by `customCode` overrides the spelling dictionary, to allow for non-standard languages such as the medical dictionaries provided with Spell Checker Pro. For all languages, ensure that:
 
 * The `code` property is valid language tag, according to https://www.ietf.org/rfc/bcp/bcp47.txt[Tags for Identifying Languages (BCP47)]
-* Either the `code` property is one of the <<supportedlanguages,Supported Languages>> or the `customCode` override property is defined.
-* The `customCode` property is one of the <<supportedlanguages,Supported Languages>> if it is defined.
+* Either the `code` property is one of the xref:supportedlanguages[Supported Languages] or the `customCode` override property is defined.
+* The `customCode` property is one of the xref:supportedlanguages[Supported Languages] if it is defined.
```
Comment (lnewson) [inline modules/ROOT/partials/plugins/tinymcespellchecker-content-langs.adoc:15]:
```suggestion
* Either the `code` property is one of the xref:plugins/premium/tinymcespellchecker.adoc#supportedlanguages[Supported Languages] or the `customCode` override property is defined.
* The `customCode` property is one of the xref:plugins/premium/tinymcespellchecker.adoc#supportedlanguages[Supported Languages] if it is defined.
```

File: modules/ROOT/partials/plugins/tinymcespellchecker-content-langs.adoc [hunk 9]
```diff
@@ -11,8 +11,8 @@ include::partial$configuration/content_langs_base.adoc[]
 The Spell Checker Pro plugin will use the `code` property to determine which dictionary to use on the content, unless the `customCode` property is available. The language tag specified by `customCode` overrides the spelling dictionary, to allow for non-standard languages such as the medical dictionaries provided with Spell Checker Pro. For all languages, ensure that:
```
Comment (lnewson) [inline modules/ROOT/partials/plugins/tinymcespellchecker-content-langs.adoc:11]: This file needs an id added to the option title. e.g:
```
[[content_langs]]
=== `content_langs`
```

**Review by lnewson — CHANGES_REQUESTED** [review]

File: modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc (hunk 2 above)
Comment (hamza0867) [reply modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc:82]: Done

File: modules/ROOT/partials/configuration/spellchecker_ignore_list.adoc (hunk 3 above)
Comment (hamza0867) [reply modules/ROOT/partials/configuration/spellchecker_ignore_list.adoc:4]: Done

File: modules/ROOT/partials/configuration/spellchecker_select_languages.adoc (hunk 6 above)
Comment (hamza0867) [reply modules/ROOT/partials/configuration/spellchecker_select_languages.adoc:5]: Done

File: modules/ROOT/partials/plugins/tinymcespellchecker-content-langs.adoc (hunk 9 above)
Comment (hamza0867) [reply modules/ROOT/partials/plugins/tinymcespellchecker-content-langs.adoc:13]: Done

**Comment by hamza0867** [comment]
> There's a broken `include` in the `content_langs` section on this page (see `modules/ROOT/partials/plugins/tinymcespellchecker-content-langs.adoc`): image
> 
> * [ ]  Fix broken include

Done

File: modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc [hunk 10]
```diff
@@ -77,10 +77,6 @@ include::partial$plugins/tinymcespellchecker-content-langs.adoc[]
```
Comment (lnewson) [inline modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc:77]: I can't comment on it sorry, but the `misc/spellchecker-languages.adoc` line above has the wrong heading level which is causing an issue with the ToC. It needs to be this instead:
```
include::partial$misc/spellchecker-languages.adoc[leveloffset=+2]
```

**Review by lnewson — APPROVED** [review]

**Review by EkimChau — APPROVED** [review]

File: modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc (hunk 10 above)
Comment (hamza0867) [reply modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc:77]: Done

Comment (hamza0867) [reply modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc:77]: @lnewson Hi, can you please resolve this conversation if the changes made are correct ? Thank you

Comment (lnewson) [reply modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc:77]: Could have sworn I resolved this sorry @hamza0867.

Comment (hamza0867) [reply modules/ROOT/pages/plugins/premium/tinymcespellchecker.adoc:77]: No worries 😄

**Review by metricjs — APPROVED** [review]

**Pull request merged**